### PR TITLE
Have speedseq per-lane alignment header only include matching instrument data's RG

### PR DESCRIPTION
**Pull request by susannasiebert**
Previously the per-lane alignment header would just be the same as the merged alignment header. Now it will only contain the @RG tag for the matching instrument data.

This results in certain QC tools not outputting multiple metrics for each RG group so the library name check can be removed.

### CODE REVIEW

**Comment by susannasiebert** [comment]
This includes all the commits from #796. I will rebase once 796 has been merged.

**Comment by susannasiebert** [comment]
I rebased this PR so it's now clean and ready to be reviewed.

**Comment by acoffman** [comment]
+1

It was initially a little confusing to see `prepare_scratch_sam_file` get called and then the result treated like the header, but then I saw that you've overridden `scratch_sam_file_path` to just return the header. It probably indicates that these methods need better names, but we don't really want to break API compatibility with other aligners, so I guess we're good for now!

File: lib/perl/Genome/InstrumentData/AlignmentResult/Speedseq.pm [hunk 1]
```diff
@@ -61,13 +63,18 @@ sub _inititalize_revivified_bam {
...
 
-#Use merged bam for header since we don't have an original per-lane bam
-sub source_bam_path_for_header {
+sub create_bam_header {
     my $self = shift;
-    return $self->get_merged_bam_to_revivify_per_lane_bam;
+
+    return $self->bam_header_path if -s $self->bam_header_path;
+
+    $self->prepare_scratch_sam_file;
```
Comment (dufeiyu) [inline lib/perl/Genome/InstrumentData/AlignmentResult/Speedseq.pm]: Regard to Adam's comment, an option here is to extract line 591 - 616 of AlignmentResult.pm to a new method prepare_scratch_sam_header_file, and then use $self->prepare_scratch_sam_header_file here.

Comment (susannasiebert) [inline lib/perl/Genome/InstrumentData/AlignmentResult/Speedseq.pm]: I like that solution. I just pushed a commit for it.

**Comment by dufeiyu** [comment]
+1

**Pull request merged**